hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java

```diff
@@ -665,8 +665,7 @@ private IdHolder doIndexQueryBatch(IndexLabel indexLabel,
 
                 // Iterate one batch, and keep iterator position
                 Set<Id> ids = InsertionOrderUtil.newSet();
-                while ((ids.size() < batch || batch == Query.NO_LIMIT) &&
-                       entries.hasNext()) {
+                while (ids.size() < batch && entries.hasNext()) {
```

Comment:
to check

Reply:
we should revert the condition indeed between `||` , updated